Update building & running doc (#2801)
magda-io/magda: docs/docs/building-and-running.md

-   [Java 8 JDK](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) - To run the JVM components, and to build the small amount of Java code.
-   [sbt](http://www.scala-sbt.org/) - To build the Scala components.
-   [yarn](https://yarnpkg.com/) - Npm replacement that makes node deps in a monorepo much easier.

To push the images and run them on kubernetes, you'll need to install:

-   [GNU tar](https://www.gnu.org/software/tar/) - (Mac only) MacOS ships with `BSD tar`. However, you will need `GNU tar` for docker images operations. On MacOS, you can install `GNU Tar` via [Homebrew](https://brew.sh/): `brew install gnu-tar`
-   [gcloud](https://cloud.google.com/sdk/gcloud/) - For the `kubectl` tool used to control your Kubernetes cluster. You will also need to this to deploy to our test and production environment on Google Cloud.

> -   [Helm 3](https://helm.sh/docs/intro/install/) to manage kubernetes deployments and config. Since version `0.0.57`, Magda requires helm 3 to deploy.

Reviewer: ```suggestion
-   [Helm 3](https://helm.sh/docs/intro/install/) to manage kubernetes deployments and config. Magda `0.0.57` and higher requires helm 3 to deploy.
```

First clone the magda directory and `cd` into it.

Then install dependencies and set up the links between components by running:

```bash
yarn install
```

Once the above prerequisites are in place, and the dependencies are installed, building MAGDA is easy.
From the MAGDA root directory, simply run the appropriate build command:

```bash
# If using lerna v3.18.0 or higher
yarn lerna run build --stream --concurrency=1 --include-dependencies

> ```

Reviewer: ```suggestion
# Otherwise:
yarn lerna run build --stream --concurrency=1 --include-filtered-dependencies
```
```

Now you can build the docker containers locally - this might take quite a while so get a cup of tea.

```bash
eval $(minikube docker-env) # (If you're running in minikube and haven't run this already)
yarn lerna run build --stream --concurrency=1 --include-dependencies # (if you haven't run this already)

> yarn lerna run docker-build-local --stream --concurrency=1 --include-filtered-dependencies

Reviewer: I think that's the newer lerna switch?
```suggestion
yarn lerna run docker-build-local --stream --concurrency=1 --include-dependencies
```

Author: @sajidanower23  I upgraded the lerna version in package so users should get `3.20.2` when they run `yarn install`

Reviewer: Yep, and lerna 3.18 or above requires `--include-filtered-dependencies`, does it not?

```bash
eval $(minikube docker-env) # (If you haven't run this already)

> yarn lerna run build --stream --concurrency=1 --include-dependencies # (if you haven't run this already)
> yarn lerna run docker-build-local --stream --concurrency=1 --include-filtered-dependencies

Reviewer: `--include-filtered-dependencies` is to be used for lerna ^3.18.0
```suggestion
yarn lerna run build --stream --concurrency=1 --include-filtered-dependencies # (if you haven't run this already)
yarn lerna run docker-build-local --stream --concurrency=1 --include-filtered-dependencies
# If you are using lerna 3.18.0 or higher, run these instead:
yarn lerna run build --stream --concurrency=1 --include-dependencies # (if you haven't run this already)
yarn lerna run docker-build-local --stream --concurrency=1 --include-dependencies
```

Author: lerna 3.18.0 has been released 6 months ago.
How about put version requirement in `Prerequisites` section on the top to avoid making our document even more complex to follow?
@AlexGilleran any ideas?

Reviewer: I think that's a great idea :+1:

Reviewer 2: :+1: :+1:

Reviewer 2: Better still would be to just remove lerna as a dependency from the docs completely, and make it so every `lerna` command is through `yarn` and uses the version installed by `yarn` (which I think we might have already done in this PR?)
